java/maven.embedder/external/binaries-list

```diff
@@ -17,3 +17,4 @@
 DC15DFF8F701B227EE523EEB7A17F77C10EAFE2F org.jdom:jdom2:2.0.6.1
 5D9CE6ADD7B714B8095F0E3E396C5E9F8C5DCFEF org.apache.maven.shared:maven-dependency-tree:2.2
 FC5C01A07E4A2DDF84AF0DFADF382E6F7993462D org.apache.maven:apache-maven:3.9.6:bin@zip
+C4A06A64E650562F30B7BF9AAEC1BFED43ACA12B com.google.guava:failureaccess:1.0.2
```

Comment:
i believe `failureaccess` is a dependency of guava which is part of the maven distribution zip from the line above. Is there a reason to upgrade a single transitive dependency from that maven distribution bundle? Keep in mind maven was not tested with that library version.

it is likely that the next maven point release will update it automatically - they typically keep their dependencies up to date.

Reply:
I hope I can say there's a reason to upgrade the library in the distribution - and it does not really matter if it originates transitively in some other distribution. I don't think there's any significant difference between the versions (1.0.2 has an Automatic-Module-Name, that's the only thing I could find), but, in principle, even the file name could be problematic, I agree.

That said - I'm fine with not merging this, I just think it is better to have a PR open against apache/netbeans (which can then be used by the project), than just having this buried inside a external patch queue.

Reply:
lets take a look at the first level dependencies:
NB 21 uses guava-32.1.2-jre right now, the pom of it is here:
https://repo1.maven.org/maven2/com/google/guava/guava/32.1.2-jre/guava-32.1.2-jre.pom

Looking at the pom, guava 32.1.2 was tested and released with failureaccess 1.0.1.

To upgrade the failureaccess lib wrapper, I would recommend to update the guava lib wrapper itself to a version which depends on the desired failureaccess version (1.0.2).
that would be 33.0.0-jre.
https://repo1.maven.org/maven2/com/google/guava/guava/33.0.0-jre/guava-33.0.0-jre.pom
So the guava lib wrapper 33.0.0 and failureaccess 1.0.2 lib wrapper would be updated in one atomic commit to versions which fit together.

further, I think we should leave `java/maven.embedder` and gradle alone, since the update will automatically happen when the maven/gradle distribution updates.

I think there are no other failureaccess users outside of guava - at least I can't find any. It is also part of the guava repo and no separate project.

Reply:
Purely regarding "Looking at the pom, guava 32.1.2 was tested and released with failureaccess 1.0.1." - the OSGi metadata allow `[1.0,2)`, i.e. `1.0.2` is OK. And there are no real differences between 1.0.1 and 1.0.2, except for the Automatic-Module-Name to the best of my knowledge.

I agree that we'll upgrade the whole Guava at some point, although that may be a tiny bit more difficult (or not - but at least the license must be re-checked thoroughly). But until Maven is solved, this is a bit unimportant for my purpose - we would still have the failureaccess 1.0.1, and it does not really matter it is part of "Maven". (It may still be fine for other purposes, of course.)

That said - I opened this mostly because I didn't want this to just be buried in some external patch queue (out of which it would be effectively impossible to adopt), setting up an unfortunate precedent along the way.

Reply:
@lahodaj I can take a look this weekend if we can update guava and prepare a PR. Will ping you if it succeeds. I am sure we will find a way to update all the relevant libs for NB 22 so that it is synced again with the javavscode project.

cc @matthiasblaesing @neilcsmith-net

Reply:
please see https://github.com/apache/netbeans/pull/7004

Reply:
The big thing missing to me here, is, why?